src/server/utils/user.js

```diff
@@ -0,0 +1,68 @@
+/* Return a new user object based on fields.
+ *
+ * This assumes that the field set was already mapped through mapColumns(...,
+ * 'auth'). It is mostly of use to the login manager; afterward, the object
+ * should be stored as req.session.user.
+ *
+ * As this is stored in the session, ensure that keys and values stored here
+ * are serializeable. In particular, ensure this object can losslessly be
+ * passed via JSON.
+ */
+function newUser(fields) {
+  // Shallow copy the user object to avoid reference woes.
+  // NB: Don't put any objects in here that you don't want sharing references.
+  // Arrays are special-cased.
+  var user = {};
+  Object.keys(fields).map(key => {
```

Comment:
If you're looking to make a shallow copy you  just do something like the following:

```js
const user = {
  ...fields
};
```

Reply:
Also functional is

```javascript
let user = Object.assign({}, fields);
```

(Note that user gets mutated quite a few times)

...but I chose this formulation specifically to `slice` the arrays so there wouldn't be any accidental reference issues. The comment alludes to exactly that :P